packages/@react-spectrum/s2/src/Card.tsx

```diff
@@ -0,0 +1,669 @@
+        // @ts-ignore - React < 19 compat
+        inert={isSkeleton ? 'true' : undefined}
+        ref={domRef}
+        className={card({size, density, variant, isCardView: ElementType !== 'div'})}>
+        <InternalCardContext.Provider value={{size, isQuiet, isHovered: false, isFocusVisible: false, isSelected: false}}>
+          {children}
+        </InternalCardContext.Provider>
+      </div>
+    );
+  }
+
+  let press = pressScale(domRef, UNSAFE_style);
+  return (
+    <ElementType
+      {...props}
+      ref={domRef}
+      className={renderProps => UNSAFE_className + card({...renderProps, isCardView: true, isLink: !!props.href, size, density, variant}, styles)}
+      style={renderProps => 
+        // Only apply press scaling to card when it has an action, not selection.
+        // When clicking the card selects it, the checkbox will scale down instead.
+        // TODO: confirm with design
+        // @ts-ignore - do we want to expose hasAction publically in RAC?
+        renderProps.hasAction && (renderProps.selectionMode === 'none' || renderProps.selectionBehavior === 'toggle') 
+          ? press(renderProps) 
+          : UNSAFE_style
+      }>
+      {({selectionMode, selectionBehavior, isHovered, isFocusVisible, isSelected, isPressed}) => (
+        <InternalCardContext.Provider value={{size, isQuiet, isHovered, isFocusVisible, isSelected}}>
+          {!isQuiet && <SelectionIndicator />}
+          {selectionMode !== 'none' && selectionBehavior === 'toggle' && (
+            <PressResponder isPressed={isPressed}>
+              <div 
+                className={style({
+                  position: 'absolute',
+                  top: 8,
+                  zIndex: 2,
+                  insetStart: 8,
+                  padding: '[6px]',
+                  backgroundColor: lightDark('transparent-white-600', 'transparent-black-600'),
+                  borderRadius: 'default',
+                  boxShadow: 'emphasized'
+                })}>
+                <Checkbox
+                  slot="selection"
+                  excludeFromTabOrder />
+              </div>
+            </PressResponder>
+          )}
+          {/* this makes the :first-child selector work even with the checkbox */}
+          <div className={style({display: 'contents'})}>
+            {children}
+          </div>
+        </InternalCardContext.Provider>
+      )}
+    </ElementType>
+  );
+});
+
+function SelectionIndicator() {
+  let {isSelected} = useContext(InternalCardContext);
+  return <div className={selectionIndicator({isSelected})} />;
+}
+
+export interface CardPreviewProps extends UnsafeStyles {
+  children: ReactNode
+}
+
+export const CardPreview = forwardRef(function CardPreview(props: CardPreviewProps, ref: DOMRef<HTMLDivElement>) {
+  let {size, isQuiet, isHovered, isFocusVisible, isSelected} = useContext(InternalCardContext);
+  let {UNSAFE_className, UNSAFE_style} = props;
+  let domRef = useDOMRef(ref);
+  return (
+    <div
```

Comment:
should we add filterDOMProps to all of these, people always want to pass their data-*s